### PR TITLE
🧬 [Oasis Create Data] Create cache system + Aave V2 handler

### PR DESCRIPTION
**Pull request by marcinciarka**
# [[Oasis Create Data] Create cache system](https://app.shortcut.com/oazo-apps/story/9098/oasis-create-data-create-cache-system)
# [[Oasis Create Data] Create handler for Aave V2](https://app.shortcut.com/oazo-apps/story/9495/oasis-create-data-create-handler-for-aave-v2)

  
## Changes 👷‍♀️
- enforced `LendingProtocol` usage everywhere
- changes to `LendingProtocolLabel`
- changes to `AssetsTableRowData` regarding the db schema
- pulling db data, validations on `/api/product-hub`
- mocking db data method
- updating data methods
- some basic updating db stuff
- aave v2 handler
  
## How to test 🧪
- enable the feature
- go to `/oasis-create/borrow`
- there should be _nothing_
- start the db again with a clear state
- migration should work OOTB with launch.sh but you can run it manually using `yarn migrate`
- use the mocks to load up the db with mock data
- oasis create page should be populated with data now

### CODE REVIEW

File: handlers/product-hub/helpers/index.ts
Comment (piotrkonowrocki) [inline handlers/product-hub/helpers/index.ts:1]: Shouldn't each helper be each own file and index just export them? Isn't that what we decided on code conventions meetings or after?

File: handlers/product-hub/index.ts [hunk 1]
```diff
@@ -1,26 +1,116 @@
...
+  // this mocks the data using a static file mock
+  // and creates necessary rows in the db (careful with this)
+  // add this to the handler
+  // case 'PUT':
+  //   return await mockProductHubData(req, res)
+  // and use postman to send a PUT request to the endpoint
+  await prisma.productHubItems
+    .createMany({
+      // @ts-ignore
+      data: mockData.table.map(({ tooltips, ...item }) => ({
```
Comment (piotrkonowrocki) [inline handlers/product-hub/index.ts:90]: What is happening here so `@ts-ignore` is needed?

File: pages/api/product-hub.tsx [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
   switch (req.method) {
     case 'POST':
-      return await handleProductHubData(req, res)
-    case 'UPDATE':
-      return await updateProductHubData(req, res)
-    case 'GET':
+      return handleGetProductHubData(req, res)
+    case 'PATCH':
+      return updateProductHubData(req, res)
+    case 'PUT':
```
Comment (piotrkonowrocki) [inline pages/api/product-hub.tsx:15]: What is the point of this method and why is it a `PUT` method? If we still need ability to get mock data via API maybe we need another endpoint?

File: handlers/product-hub/helpers/index.ts
Comment (marcinciarka) [reply handlers/product-hub/helpers/index.ts:1]: Done.

File: handlers/product-hub/index.ts (hunk 1 above)
Comment (marcinciarka) [reply handlers/product-hub/index.ts:142]: Don't remember, removed.

File: pages/api/product-hub.tsx (hunk 2 above)
Comment (marcinciarka) [reply pages/api/product-hub.tsx]: Mock is a separated endpoint now.

File: scripts/docker/compose.yml [hunk 3]
```diff
@@ -13,22 +13,22 @@ services:
       - '5432:5432'
     volumes:
       - db_data:/var/lib/postgresql/data
-  localnode:
```
Comment (piotrkonowrocki) [inline scripts/docker/compose.yml]: I assume this is not intentional, right?

File: scripts/docker/docker-compose.yaml [hunk 4]
```diff
@@ -11,20 +11,20 @@ services:
...
-      - PRIV_KEY_MAINNET=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80
-      - TENDERLY_USERNAME=georgi2
-      - TENDERLY_PROJECT=mpoc
-      - USE_DUMMY=${USE_DUMMY}
-      - BLOCK_NUMBER=${BLOCK_NUMBER}
-    ports:
-      - '8545:8545'
-    volumes:
-      - '../.hardhat/cache:/usr/src/mpa/cache'
+  # localnode:
```
Comment (piotrkonowrocki) [inline scripts/docker/docker-compose.yaml]: Same as above.

File: handlers/product-hub/index.ts [hunk 5]
```diff
@@ -71,13 +84,63 @@ export async function updateProductHubData(
...
-    return { name, data: call() }
-  })
-  return res.status(200).json({ data })
+  const dataHandlersPromiseList = await Promise.all(
+    handlersList.map(({ name, call }) => {
+      const startTime = Date.now()
+      return call().then((data) => ({
+        name, // protocol name
+        data,
+        processingTime: measureTime ? Date.now() - startTime : undefined,
```
Comment (piotrkonowrocki) [inline handlers/product-hub/index.ts:93]: If you set `measureTime` to `false`, is it going to return `undefined` or skip this prop at all? Shouldn't it be like that?
```typescript
...(measureTime && {
  processingTime: Date.now() - startTime
})
```

**Review by piotrkonowrocki — APPROVED** [review]

**Pull request merged**